infra/modules/providers/azure/service-principal/sample/main.tf

```diff
@@ -0,0 +1,60 @@
+//  Copyright © Microsoft Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+provider "azurerm" {
+  version = "=1.44.0"
```

Comment:
Any common way to specify this so when we move to newer version, we change only at one place?

Reply:
There is yes and we do.  https://github.com/Azure/osdu-infrastructure/blob/master/infra/templates/osdu-r2-resources/commons.tf#L15  

This however is a small sample used as a unit test where I'm specifically calling out that provider version that I wanted to test against.